openedx/openedx-platform
# WIP: LX-116 Converts Video XModule to an XBlock

WIP: Initial move of video_module code into xblock_builtin/xblock_video

* Video XBlock loads in LMS with original Video XModule OLX and separate video definition files.
* Videos are no longer exported to a separate `videos` definition files; they're inlined in the vertical.
* Both `video` and legacy `videoalpha` types supported (see Demo Course)

**JIRA tickets**: [OSPR-2830](https://openedx.atlassian.net/browse/OSPR-2830) LX-116

**Dependencies**: 

* https://github.com/edx/XBlock/pull/419

**Screenshots**: N/A -- no UI changes expected.

**Sandbox URL**
* LMS: https://pr19303.sandbox.opencraft.hosting/
* Studio: https://studio-pr19303.sandbox.opencraft.hosting/

(f22baec -- redeploying)

**Note** We're experiencing a bug with authentication for our sandboxes based on the latest master, and so to use the sandbox, you may need to work around this issue by logging into the Django Admin instead:
* LMS: https://pr19303.sandbox.opencraft.hosting/admin
* Studio: https://studio-pr19303.sandbox.opencraft.hosting/admin

**Testing instructions**

Setup notes:
* You'll need to run the following to use this branch on your devstack:
   ```
   make lms-static
   make studio-static 
   ```

Testing:
1. Import the [edx Demo course](https://github.com/edx/edx-demo-course/releases/tag/open-release%2Fhawthorn.2).
1.  Ensure that the videos in the Demo course are visible, specifically:
    * [Introduction: Video and Sequences](https://preview-pr19303.sandbox.opencraft.hosting/courses/course-v1:OpenCraft+DemoX+2018_1/courseware/d8a6192ade314473a78242dfeedfbf5b/edx_introduction/1?activate_block_id=block-v1%3AOpenCraft%2BDemoX%2B2018_1%2Btype%40vertical%2Bblock%40vertical_0270f6de40fc) -- `videoalpha` block type
    * [ Videos on edX](https://pr19303.sandbox.opencraft.hosting/courses/course-v1:edX+DemoX+Demo_Course/courseware/interactive_demonstrations/19a30717eff543078a5d94ae9d6c18a5/1?activate_block_id=block-v1%3AedX%2BDemoX%2BDemo_Course%2Btype%40vertical%2Bblock%40867dddb6f55d410caaa9c1eb9c6743ec) -- `video` block type, shows transcript.
1. Enable completion tracking by activating the `completion.enable_completion_tracking` waffle flag.
    Ensure that watching videos to the end triggers a `publish_completion` event (404s, see notes)
1. Export the Demo course and compare the export with the original [edx Demo course](https://github.com/edx/edx-demo-course/releases/tag/open-release%2Fhawthorn.2).
   
**Author Notes & Concerns**
1. Viewing, adding, and editing video blocks in Studio is not working yet.
1. Subtitles and transcripts have not been tested extensively, so some issues may remain with imported/exported courses.
1. This initial PR is a prototype, and so most code has been moved with minimal modifications.
  Code refactoring and cleanup and test fixes are expected before this PR will be ready for approval.
1. Completion event handling is not working yet.

**Reviewers**
- [ ] edX reviewer[s] TBD -- FYI @ormsbee
